crates/pink/sidevm/env/src/tasks.rs

```diff
@@ -19,11 +20,57 @@ thread_local! {
     /// New spawned tasks are pushed to this queue. Since tasks are always spawned from inside a
     /// running task which borrowing the TASKS, it can not be immediately pushed to the TASKS.
     static SPAWNING_TASKS: RefCell<Vec<TaskFuture>> = RefCell::new(vec![]);
+    /// Wakers might being referenced by the sidevm host runtime.
+    ///
+    /// When a ocall polling some resource, we can not pass the waker to the host runtime,
+    /// because they are in different memory space and in different rust code compilation space.
+    /// So when we poll into the host runtime, we cache the waker in WAKERS, and pass the index,
```

Comment:
Should we mention the negative waker id means to be dropped?

Reply:
Should